libs/s25main/ingameWindows/iwBuildingProductivities.cpp

```diff
@@ -25,9 +29,27 @@ const std::array<BuildingType, 24> iwBuildingProductivities::icons = {
   BuildingType::Farm,          BuildingType::CoalMine,
   BuildingType::PigFarm,       BuildingType::GraniteMine,
   BuildingType::DonkeyBreeder, BuildingType::Charburner,
+  BuildingType::Vineyard,      BuildingType::Winery,
+  BuildingType::Temple,
   // clang-format on
 };
 
+void iwBuildingProductivities::setBuildingOrder()
+{
+    std::copy(std::begin(allIcons), std::end(allIcons), std::back_inserter(usedIcons));
+    removeUnusedBuildings();
+}
+
+void iwBuildingProductivities::removeUnusedBuildings()
+{
+    auto removeNotUsedBuilding = [=](BuildingType const& bts) {
+        return ((!wineaddon::isAddonActive(player.GetGameWorld()) && wineaddon::isWineAddonBuildingType(bts))
+                || (!player.GetGameWorld().GetGGS().isEnabled(AddonId::CHARBURNER) && bts == BuildingType::Charburner));
+    };
+    usedIcons.erase(std::remove_if(std::begin(usedIcons), std::end(usedIcons), removeNotUsedBuilding),
+                    std::end(usedIcons));
```

Comment:
Combine and simplify
```suggestion
    usedIcons.assign(allIcons.begin(), allIcons.end()
    const auto isUnused = [&](BuildingType const& bld) {
        if (!wineaddon::isAddonActive(player.GetGameWorld() && wineaddon::isWineAddonBuildingType(bld))
            return false;
        if(!player.GetGameWorld().GetGGS().isEnabled(AddonId::CHARBURNER) && bld == BuildingType::Charburner));
            return false;
        return true;
    };
    helpers::erase_if(usedIcons, isUnused);
```

Basically:
- Rename condition lambda from action verb to adjective
- the lambda needs to capture by reference or you'll copy at least `player`
- Split visibly into the 3 cases (avoids the prior confusion, did clang-tidy suggest to combine those? in that case keep it of course. But I guess it was just the `else` in `if-return-else`)
- for most container ops requiring a range over the full container there is a helper to avoid mistakes when repeating the container name. Kind of like ranges.
- Single use function can be inlined, there is no other possible use for this as you'd always need to refill the building first. With the short lambda and `helpers::erase_if(usedIcons, isUnused);` it's pretty clear what this code does already so the function name in this case doesn't help much for readability.

Reply:
Applied the changes at all places